src/server/transaction.cc

```diff
@@ -966,6 +968,16 @@ const absl::flat_hash_set<std::string_view>& Transaction::GetMultiKeys() const {
   return multi_->frozen_keys_set;
 }
 
+void Transaction::FIX_ConcludeJournalExec() {
+  if (!multi_->shard_journal_write.front())
+    return;
+
+  if (auto journal = EngineShard::tlocal()->journal(); journal != nullptr) {
+    journal->RecordEntry(txid_, journal::Op::EXEC, db_index_, 1,
+                         unique_slot_checker_.GetUniqueSlotId(), {}, true);
```

Comment:
the last param means that you can preempt when this is executed. I think its safer if we do not preempt here as this function runs inside a callback on the tx queue.
Now when reviewing the flow I see we can preempt when running a squeshed transaction as we call log to journal there between different callbacks. I cant say for sure we dont have bugs there